limbo: fix commit/rollback failures with triggers (#8736)
tarantool/tarantool: changelogs/unreleased/gh-8505-synchro-triggers-fail.md

## bugfix/core

> * Fixed a bug because of which transactions on synchronous spaces failed with
>   the ER_CURSOR_NO_TRANSACTION error, when on_commit/on_rollback triggers were
>   set (gh-8505).

Reviewer: ```suggestion
* Fixed a bug causing the `ER_CURSOR_NO_TRANSACTION` failure for transactions
  on synchronous spaces when the `on_commit/on_rollback` triggers are
  set (gh-8505).
```
